holoviews/core/data/__init__.py

```diff
@@ -68,25 +68,38 @@ class DataConversion(object):
     def __init__(self, element):
         self._element = element
 
-    def __call__(self, new_type, kdims=None, vdims=None, mdims=None,
+    def __call__(self, new_type, kdims=None, vdims=None, groupby=None,
                  sort=False, **kwargs):
         """
-        Generic conversion method for Column based types. Supply the
-        Columns based type to convert to and optionally the
-        key dimensions (kdims), value dimensions (vdims) and HoloMap
-        key dimensions (mdims). Converted Columns can be automatically
-        sorted via the sort option and kwargs can be passed through.
+        Generic conversion method for Dataset based Element
+        types. Supply the Dataset Element type to convert to and
+        optionally the key dimensions (kdims), value dimensions
+        (vdims) and the dimensions.  to group over. Converted Columns
+        can be automatically sorted via the sort option and kwargs can
+        bepassed through.
         """
+        if 'mdims' in kwargs:
+            if groupby:
+                raise ValueError('Cannot supply both mdims and groupby')
+            else:
+                self._element.warning("mdims keyword has been renamed "
+                                      "to groupby; the name mdims is "
+                                      "deprecated and will be removed "
+                                      "after version 1.7.")
```

Comment:
I never know whether it should be ``'mdims' keyword ...`` or ``mdims keyword...``. As we are referring to the name of the keyword, I think I would include the quotes. Anyway, very minor issue and I don't mind if you prefer to leave it as it is.